effects/ox/src/main/scala/sttp/client4/impl/ox/ws/OxWebSockets.scala

```diff
@@ -0,0 +1,117 @@
+package sttp.client4.impl.ox.ws
+
+import ox.*
+import ox.channels.*
+import sttp.client4.ws.SyncWebSocket
+import sttp.ws.WebSocketFrame
+
+import scala.util.control.NonFatal
+
+/** Converts a [[SyncWebSocket]] into a pair of `Source` of server responses and a `Sink` for client requests. The
```

Comment:
maybe let's use "messages" instead of request-response? With websockets, it's hard to tell which are requests, which are responses ;)